### DISCUSSION TITLE
Deploy my portfolio on Vercel has  build failed

### DISCUSSION THREAD

**Discussion by HenriqueSerres** [vercel/next.js#54904] [Help]
### Summary

Error: Module not found: Can't resolve 'encoding' in '/vercel/path0/node_modules/node-fetch/lib'
I run `npm i -D enconfig`, but don´t works

Error: Missing API key. Pass it to the constructor `new Resend("re_123")`
I send .env folder to github, but don´t work.

Error: Failed to collect page data for /api/send

### Example

https://github.com/HenriqueSerres/my-portfolio

**Comment by devjiwonchoi** [comment]
1. `encoding` does not affect build error, it is just a warning.
2. Do not reveal `.env.local` to the public, use [Vercel Environment Variables](https://vercel.com/docs/projects/environment-variables). On development, use `.env.(test | development | production).local` if necessary.
3. When using `fetch`, provide the full URL (e.g. `http://localhost:3000`), I'd recommend putting this also in env.

```env
# .env.development.local
API_URL=http://localhost:3000/api
```
```ts
fetch(`${process.env.API_URL}/send`)
```

### Reference
- [Environment Variables](https://nextjs.org/docs/app/building-your-application/configuring/environment-variables)

**Comment by arfin-arif** [comment]
in my case, it worked when add NEXT_PUBLIC_ for the env variables

**Reply by faustinozanetto** [reply]
But you are revealing a sensitive Resend Api Key

**Reply by kkonstnar** [reply]
how should we do it ?

**Reply by faustinozanetto** [reply]
> how should we do it ?
> 
> 

I moved the resend create inside the api body and it seemed to have fixed it

**Comment by kaushal7075** [comment]
> ### Summary
> Error: Module not found: Can't resolve 'encoding' in '/vercel/path0/node_modules/node-fetch/lib' I run `npm i -D enconfig`, but don´t works
> 
> Error: Missing API key. Pass it to the constructor `new Resend("re_123")` I send .env folder to github, but don´t work.
> 
> Error: Failed to collect page data for /api/send
> 
> ### Additional information
> _No response_
> 
> ### Example
> https://github.com/HenriqueSerres/my-portfolio

how you solve this issue?

**Comment by ruihuaniu** [comment]
I got the same issue, but the fix is simple, just delete and re-add the RESEND_API_KEY key in Vercel, and then redeploy the project.

**Reply by ranyhakley** [reply]
This worked for me, i deployed my project before adding my key. I should've done it the other way around!
